bundles/io/org.eclipse.smarthome.io.http.auth.form/src/main/java/org/eclipse/smarthome/io/http/auth/form/internal/LoginService.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.http.auth.form.internal;
+
+import org.eclipse.smarthome.io.http.HttpContextFactoryService;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.http.HttpService;
+import org.osgi.service.http.NamespaceException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A login service which is responsible for registering a form processing endpoint.
+ *
+ * @author Łukasz Dywicki - Initial contribution and API.
+ */
+@Component
+public class LoginService {
+
+    private static final String LOGIN_FORM_ADDRESS = "/login/form";
```

Comment:
This should be "/login", so that the page is available as `/login/index.html`.
It might even be more elegant to register a servlet under `/login`, which simply serves the page.

Reply:
Just noticed that the `?process` is anyhow not processed by the form, so I assume you'll have to make this a servlet anyhow, so that you can set the Basic Auth HTTP header, right?